test/test-platform/TestPlatform.cpp

```diff
@@ -289,153 +290,124 @@ CTestPlatform::CommandReturn CTestPlatform::applyConfigurations(const IRemoteCom
+    for (uint32_t state = 0; state < nbStates; state++) {
 
-        const std::string& strValue = remoteCommand.getArgument(uiState + 1);
-
-        if (!pCriterionType->addValuePair(
-                    0x1 << uiState, strValue, strResult)) {
-
-            strResult = "Unable to add value: " + strValue + ": " + strResult;
+        const std::string& value = remoteCommand.getArgument(state + 1);
 
+        if (values.count(value) != 0) {
+            result = "Unable to add value: " + value + ": because it's referenced twice";
             return false;
         }
+        values[value] = state + 1;
     }
 
-    _pParameterMgrPlatformConnector->createSelectionCriterion(strName, pCriterionType);
+    CriterionInterface* criterion =
+        _pParameterMgrPlatformConnector->createInclusiveCriterion(name, values, result);
+
+    if (criterion == nullptr) {
+        return false;
+    }
 
     return true;
 }
 
 
-bool CTestPlatform::createExclusiveSelectionCriterion(const string& strName,
-                                                      uint32_t uiNbStates,
-                                                      string& strResult)
+bool CTestPlatform::createExclusiveCriterion(const string& name, uint32_t nbStates, string& result)
 {
-    ISelectionCriterionTypeInterface* pCriterionType =
-        _pParameterMgrPlatformConnector->createSelectionCriterionType(false);
-
-    uint32_t uistate;
-
-    for (uistate = 0; uistate < uiNbStates; uistate++) {
-
-	std::ostringstream ostrValue;
+    using namespace core::selection::criterion;
+    CriterionInterface::Values values;
 
-        ostrValue << "State_";
-        ostrValue << uistate;
-
-        if (!pCriterionType->addValuePair(
-                    uistate, ostrValue.str(), strResult)) {
-
-            strResult = "Unable to add value: "
-                + ostrValue.str() + ": " + strResult;
-
-            return false;
-        }
+    for (uint32_t state = 0; state < nbStates; state++) {
+        values["State_" + std::to_string(state)] = state;
     }
+    CriterionInterface* criterion =
+        _pParameterMgrPlatformConnector->createExclusiveCriterion(name, values, result);
 
-    _pParameterMgrPlatformConnector->createSelectionCriterion(strName, pCriterionType);
+    if (criterion == nullptr) {
+        return false;
+    }
 
     return true;
 }
 
-bool CTestPlatform::createInclusiveSelectionCriterion(const string& strName,
-                                                      uint32_t uiNbStates,
-                                                      string& strResult)
+bool CTestPlatform::createInclusiveCriterion(const string& name, uint32_t nbStates, string& result)
 {
-    ISelectionCriterionTypeInterface* pCriterionType =
-        _pParameterMgrPlatformConnector->createSelectionCriterionType(true);
+    using namespace core::selection::criterion;
+    CriterionInterface::Values values;
 
-    if (uiNbStates > 32) {
+    if (nbStates > 32) {
```

Comment:
I let to test-platform the exact same bahavior. Nevertheless, other clients can have as many inclusive criterion value as they want.
The problem with test-platform is that the automated criterion creation (we only give a number) if used in tests (functional tests for example) and use a mask to name criteria. So if we change test-platform behavior, all tests must be updated.